synapse/handlers/auth.py

```diff
@@ -520,12 +520,18 @@ async def _check_auth_dict(
             res = await checker.check_auth(authdict, clientip=clientip)
             return res
 
-        # build a v1-login-style dict out of the authdict and fall back to the
-        # v1 code
-        user_id = authdict.get("user")
+        # Retrieve the user ID from the authdict
+
+        # Check for an identifier dict
+        identifier = authdict.get("identifier", {})
+        if "type" in identifier and identifier["type"] == "m.id.user":
```

Comment:
You should be able to just get it all at once here without checking.

```suggestion
        if identifier.get("type") == "m.id.user":
```

Reply:
Nice, thanks!